scripts/utils/import-metrics/import-metrics.sh

```diff
@@ -0,0 +1,199 @@
+    while [ "${to_seconds}" -gt "${from_seconds}" ]; do
+
+        # reduce offset_seconds when last query time range will be less than offset
+        if [ $((${to_seconds} - ${from_seconds})) -lt ${offset_seconds} ]; then
+            offset_seconds=$((${to_seconds} - ${from_seconds}))
+        fi
+
+        # set date range for query
+        range=$((${to_seconds} - ${offset_seconds}))
+        from=$(date -j -f "%s" "${range}" "+%Y-%m-%dT%H:%M:%SZ")
+        to=$(date -j -f "%s" "${to_seconds}" "+%Y-%m-%dT%H:%M:%SZ")
+
+        # from separate mimirtool shell execute remote-read
+        kubectl exec -n cattle-monitoring-system mimirtool --insecure-skip-tls-verify -i -t -- mimirtool remote-read export --tsdb-path ./prometheus-export --address http://rancher-monitoring-prometheus:9090 --remote-read-path /api/v1/read --to=${to} --from=${from} --selector ${selector}
+
+        # compress metrics data from export
+        kubectl exec -n cattle-monitoring-system mimirtool --insecure-skip-tls-verify -i -t -- tar zcf /tmp/prometheus-export.tar.gz ./prometheus-export
+
+        # set filename timestamp
+        ts2=$(date -j -f "%s" "${range}" "+%Y-%m-%dT%H-%M-%S")
+
+        # copy exported metrics data to timestamped tarball
+        kubectl -n cattle-monitoring-system cp mimirtool:/tmp/prometheus-export.tar.gz ./prometheus-export-${ts2}.tar.gz 1> /dev/null
+
+        # clear export data from pod 
+        kubectl exec -n cattle-monitoring-system mimirtool --insecure-skip-tls-verify -i -t -- rm -rf prometheus-export
+
+        # unpack, navigate
+        tar xf prometheus-export-${ts2}.tar.gz 1> /dev/null
+        cd prometheus-export
+
+        # aggregate tsdb
+        cp -R `ls | grep -v "wal"` ../ || printf " - No blocks to copy \n"
+        
+        # cleanup
+        cd ../
+        rm -rf prometheus-export
+
+        # increment time range by offset_seconds
+        to_seconds=$((${to_seconds} - ${offset_seconds}))
+
+        # wait
+        sleep 5
+        
+    done
+
+    # delete mimirtool pod
+    kubectl delete pod -n cattle-monitoring-system mimirtool
+
+    # output command to run prometheus graph on metrics data (locally via docker, overlapping/obsolete blocks are handled during compaction)
+    printf "\n\e[32mMetrics import complete!\e[0m\nView metrics data locally via docker:\n\n"
+    printf "docker run --rm -u \"$(id -u)\" -ti -p 9090:9090 -v ${PWD}:/prometheus rancher/mirrored-prometheus-prometheus:v2.42.0 --storage.tsdb.path=/prometheus --storage.tsdb.retention.time=1y --config.file=/dev/null \n\n"
+
+}
+
+
+# set input variables from script arguments
+process_args(){
+
+    # regex to match arguments
+    kube_regex="(.*yaml)|(.*yml)"
+    selector_regex="({.*})"
+    offset_regex="[0-9]{4}(\-){0}$"
+    date_regex=".*T.*Z"
+
+    # count date inputs
+    date_count=0
+    
+    for arg in "$@"
+    do
+        # set kubeconfig from input
+        if [[ $arg =~ ${kube_regex} ]]; then
+            export KUBECONFIG=$1
```

Comment:
This should be `export KUBECONFIG=$arg` right?

Reply:
Or are we assuming that kubeconfig is always the first arg? 🤔

Reply:
Fixed. This was leftover from the previous iteration.